tap_linkedin_ads/streams.py

```diff
@@ -0,0 +1,699 @@
+    'startAt',
+    'endAt',
+    'creative',
+    'creativeId'
+}
+
+def write_bookmark(state, value, stream_name):
+    """
+    Write the bookmark in the state corresponding to the stream.
+    """
+    if 'bookmarks' not in state:
+        state['bookmarks'] = {}
+    state['bookmarks'][stream_name] = value
+    LOGGER.info('Write state for stream: %s, value: %s', stream_name, value)
+    singer.write_state(state)
+
+def selected_fields(catalog_for_stream):
+    """
+    Get all selected fields of given streams
+    """
+    mdata = metadata.to_map(catalog_for_stream.metadata)
+    fields = catalog_for_stream.schema.properties.keys()
+
+    selected_fields_list = list()
+    # Loop through all fields of the given stream
+    for field in fields:
+        field_metadata = mdata.get(('properties', field))
+        if should_sync_field(field_metadata.get('inclusion'), field_metadata.get('selected')):
+            selected_fields_list.append(field)
+
+    return selected_fields_list
+
+def split_into_chunks(fields, chunk_length):
+    """
+    Return list of chunk_length fields for total fields.
+    Example:
+
+    Args: fields = [1, 2, 3, 4, 5], chunk_length = 2
+    Return: [[1, 2], [3, 4], [5]]
+    """
+    return (fields[x:x+chunk_length] for x in range(0, len(fields), chunk_length))
+
+def sync_analytics_endpoint(client, stream_name, path, query_string):
+    """
+    Call API for analytics endpoint and return all pages of records.
+    """
+    page = 1
+    next_url = 'https://api.linkedin.com/v2/{}?{}'.format(path, query_string)
+
+    # Loop until the last page
+    while next_url:
+        LOGGER.info('URL for %s: %s', stream_name, next_url)
+
+        data = client.get(url=next_url, endpoint=stream_name)
+        yield data
+        # Fetch next page
+        next_url = get_next_url(data)
+
+        LOGGER.info('%s: Synced page %s', stream_name, page)
+        page = page + 1
+
+def get_next_url(data):
+    """
+    Prepare and return the URL to fetch the next page of records.
+    """
+    next_url = None
+    links = data.get('paging', {}).get('links', [])
+    for link in links:
+        rel = link.get('rel')
+        if rel == 'next':
+            href = link.get('href')
+            if href:
+                # Prepare next page URL
+                next_url = 'https://api.linkedin.com{}'.format(urllib.parse.unquote(href))
+    return next_url
+
+def shift_sync_window(params, today, date_window_size, forced_window_size=None):
+    """
+    Move ahead date window by date_window_size and update params with the new date window.
+    """
+    current_end = datetime.date(
+        year=params['dateRange.end.year'],
+        month=params['dateRange.end.month'],
+        day=params['dateRange.end.day'],
+    )
+    if forced_window_size:
+        new_end = current_end + timedelta(days=forced_window_size)
+    else:
+        new_end = current_end + timedelta(days=date_window_size)
+
+    if new_end > today:
+        new_end = today
```

Comment:
```suggestion
   new_end = min(today, current_end + timedelta(days=(forced_window_size if forced_window_size else date_window_size)))
```

Reply:
Will this satisfy above code of 7 lines?

Reply:
Yes, this change will satisfy the above code of 7 lines. Updated according to this.